src/main/kotlin/assimp/BaseImporter.kt

```diff
@@ -33,7 +33,7 @@ abstract class BaseImporter {
      *  time to examine the contents of the file to be loaded for magic bytes, keywords, etc to be able to load files
      *  with unknown/not existent file extensions.
      *  @return true if the class can read this file, false if not. */
-    abstract fun canRead(file: URI, checkSig: Boolean): Boolean
+    abstract fun canRead(file: String, pIOHandler: IOSystem, checkSig: Boolean): Boolean
```

Comment:
`pIOHandler: IOSystem = ioHandler`

After moving default IOHandler into here, you can use default values so that not every user has to specify the default one every time. This will help maintain backwards compatibility.

Reply:
Any user functions should have this same idea applied